File: API/App.Model/Entities/MatchPlayer.cs
using App.Model.Entities.Common;
using App.Model.Enums;
using System.ComponentModel.DataAnnotations;

namespace App.Model.Entities
{
    public class MatchPlayer : AuditableEntity
    {
        public Match Match { get; set; }
        public Player Player { get; set; }

        [Required]
        public PlayerPosition PlayerPosition { get; set; }
    }
}

File: WebUI/app-frontend/src/app/models/player/match-player.ts
import { PlayerPosition } from "src/app/enums/player-position";
import { SimpleTeam } from "../team/simple-team.model";

export class MatchPlayer {
    public id : string;
    public playerId : string;
    public isActive : boolean;
    public team : SimpleTeam;
    public name : string;
    public middleName : string;
    public surname : string;
    public playerPosition : PlayerPosition;
}
